### PR TITLE
chore: update deps to latest versions

### PR DESCRIPTION
**Pull request by achingbrain**
I've tried to make the minimum amount of changes necessary for this, since the underlying crypto libraries only support node Buffers or BufferLists there doesn't seem a lot of point in doing lots of conversions between Uint8Arrays and Buffers.

BREAKING CHANGES:

- All deps use Uint8Arrays in place of node Buffers

### CODE REVIEW

**Comment by achingbrain** [comment]
cc @jacobheun @vasco-santos

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,19940 @@
+{
```
Comment (mpetrunic) [inline package-lock.json:1]: We use yarn, would you mind deleting this file and updating yarn.lock

Comment (achingbrain) [reply package-lock.json]: Done, my apologies!

**Comment by mpetrunic** [comment]
@achingbrain Did you intentionally leave Buffers in noise constructor?

**Comment by achingbrain** [comment]
> Did you intentionally leave Buffers in noise constructor?

Yes - the underlying `bcrypto` module is full of `Buffer.isBuffer` so you end up converting any `Uint8Array`s to `Buffer`s before passing them through.

I did original try to remove `Buffer`s entirely but it was a much more disruptive changeset and seemed like the wrong approach given you end up converting everything anyway.

**Comment by achingbrain** [comment]
@mpetrunic could this be released please?

**Comment by vasco-santos** [comment]
A note regarding releasing this. Please do not release it with `latest` tag.

While we do not have everything bubbled up to `js-libp2p` and `js-libp2p` released, we usually handle the breaking change releases by releasing it and set a `beta` tag for the release, keeping the latest as it was. We end up putting everything in `latest` at the same time to avoid people installing noise with this PR in `js-libp2p` until it is ready

**Comment by achingbrain** [comment]
@mpetrunic from the commit log it looks like you're gearing up to release this as `1.2.0` - this should really go out as `2.0.0` as the dep updates will cause multiple versions of the peer-id module to appear in people's dep trees, some of which require node Buffers and some of which require Uint8Arrays which may break things.

**Comment by mpetrunic** [comment]
Oh, was not aware of that, tnx. i dont think release pipepline supports beta tags so I will have to release it manually anyways

**Comment by mpetrunic** [comment]
https://www.npmjs.com/package/libp2p-noise/v/2.0.0-beta released with beta tag

**Comment by achingbrain** [comment]
I think @vasco-santos meant to release it with a beta npm dist tag, rather than putting it in the version number, that way we don’t need to do follow-up PRs to remove the extra specifier.

https://docs.npmjs.com/cli/dist-tag

**Comment by mpetrunic** [comment]
Yes, there is also beta tag so you can use libp2p-noise@beta

**Comment by achingbrain** [comment]
Ah, no, I meant 2.0.0 as beta, not 2.0.0-beta as beta, otherwise we have to re-PR to remove the -beta part of the version number. See how libp2p-tcp has done it, for example.

**Comment by vasco-santos** [comment]
Yeah, I meant as @achingbrain said. Sorry for not being specific enough. 
We can update it by then now.

**Comment by mpetrunic** [comment]
we have 2.0.0 now tagged with beta :slightly_smiling_face:

**Comment by achingbrain** [comment]
Perfect, thank you!

**Pull request merged**